pkg/importer/spanner/sysl.arrai

```diff
@@ -0,0 +1,128 @@
+        entity.cluster:
+            //seq.join('', entity.cluster >>
+            $`interleave_in_parent="${.interleaved_in}", interleave_on_delete="${.on_delete}"`),
+    };
+    let fk = cond {
+        entity.foreign_keys:
+            $`foreign_keys=[${//seq.join(', ',(entity.foreign_keys =>
+                \keys $`["constraint:${keys.constraint_name}","columns:${//seq.join(', ', (keys.foreign_keys => .attribute) orderby .)}"]`) orderby .)}]`,
+    };
+    let indices = cond {
+        (model.indexes where .table_name = entity.name):
+            $`indexes=[${//seq.join(', ', ((model.indexes orderby .) where .@item.table_name = entity.name) >>
+                $`"name:${.table_name}","unique:${.unique}","null_filtered:${.nullfiltered}","key_parts:${//seq.join(', ', .key_part orderby .)}${cond {.storing_col: $`","storing:${//seq.join(', ',.storing_col)}`}}${cond {.interleaved_table: $`","interleave_in:${.interleaved_table}`}}`)}"]`,
+    };
+    //seq.join(', ', ([pk, cluster, fk, indices] where .@item != ''));
+
+# appendEntityPatterns appends the entity patterns together.
+let appendEntityPatterns = \entity \model
+    cond {
+        hasEntityPattern(entity, model): $`[${entityPatterns(entity, model)}]`
+    };
+
+# appendAttributePatterns appends the attribute patterns together.
+let appendAttributePatterns = \entity \attr
+    cond {
+        hasAttributePattern(entity,attr): $`[${attributePatterns(entity, attr)}]`
+    };
+
+# typeInfo generates the type info for an attribute.
+let typeInfo = \entity \attr \type \isArray
+    let fkAttr = entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) =>
+        .reference_attribute orderby .) where .;
+    let fkTable = entity.foreign_keys => ((.foreign_keys where .attribute = attr.name) =>
+        .reference_table orderby .) where .;
+    cond {
+        fkAttr | fkTable:  //seq.join('', (fkTable => //seq.join('', .) orderby .)) ++ '.'
+            ++ //seq.join('', (fkAttr => //seq.join('', .) orderby .)),
+        isArray: $`sequence of ${type}`,
+        _: type,
+    };
+
+# transformModel translates the empty model into sysl file.
+let transformModel = \model \package
+    # sysl specification
+    # https://github.com/anz-bank/sysl/blob/master/pkg/sysl/sysl.proto
+    $`
+        ########   THIS IS AUTOGENERATED BY sysl   ########
+        ${model.schema => $`
+            ${.name}${cond {package: $`[package="${package}"]`} }:
+            ${//seq.sub('#>>>\n', '')($`
+            ${(model.entities => \entity $`
+            #>>>
+                !table ${entity.name} ${appendEntityPatterns(entity, model)}:
+                    ${entity.attributes >>
+                        $`
+                            ${.name} <: ${typeInfo(entity, ., .type ++ size(.length), .array)}${cond {.nullable:'?'}} ${appendAttributePatterns(entity, .)}` ::\i:
+                        }`
+                    )
+                    orderby .::\i:
+            }
+            `)}
```

Comment:
You shouldn't need this magic for indenting. Something like the following should work (thanks to Novan's fixes), or there's a bug:
```suggestion
                ${(model.entities => \entity $`
                    !table ${entity.name} ${appendEntityPatterns(entity, model)}:
                        ${entity.attributes >> $`
                            ${.name} <: ${typeInfo(entity, ., .type ++ size(.length), .array)}${cond {.nullable:'?'}} ${appendAttributePatterns(entity, .)}
                        `::\i:}
                `) orderby .::\i:}
```

Reply:
This doesn't use pkg/importer/writer.go to write the sysl file.

Reply:
Doesn't work even in the lastest arrai ver 158 without hack.

Reply:
comment fixed as well.

Reply:
Here's some similar code I wrote to generate Sysl from some XLSX data. The indenting works just fine with the latest version of arr.ai (though there are more 

```arrai
let sysl = $`
App[~db]:
    ${tables orderby .@row >> $`
        !table ${.table}:
            ${(.fields orderby .@row) >> $`
                ${.field}${cond {.data_type: ' ' + $`<: ${.data_type}`}}${
                    cond {.description: $`
                        :
                            @description =:
                                ${//seq.split('\n', .description) >> $`| ${.}`::\i}
                    `}
                }
            `::\i}
    `::\i\n}
`;
```

Note the workaround below was due to a bug with string templates eating leading whitespace too aggressively, but that's a small and different problem.
```
' ' + $``
```